command/results_xunit.go

```diff
@@ -207,3 +181,36 @@ func (c *xunitResults) parseAndUploadResults(ctx context.Context, conf *model.Ta
 
 	return sendTestResults(ctx, conf, logger, comm, &task.LocalTestResults{Results: tests})
 }
+
+func addTestCasesForSuite(suite testSuite, idx int, t *task.Task, tests *[]task.TestResult, logs *[]*model.TestLog, logIdxToTestIdx *[]int) {
```

Comment:
I don't understand why you're using pointers to lists. Is it just to destructively modify them? I think it's better to just return them.

Reply:
You could just not pass in tests, logs, and logIdxToTestIdx and do the appending on the other side of the function. Alternatively you could put the three into a struct and pass a pointer to that.

Reply:
Normally for private functions I find destructively modifying things to be cleaner. In this case since we need a pointer to an array it's probably worse so I changed it to a struct